tests/unit/schema/test.browser-apis.js

```diff
@@ -22,6 +35,63 @@ describe('browserApis', () => {
       )[0].split('.');
       expect(hasBrowserApi(namespace, property)).toBeTruthy();
     });
+
+    it('has not the API when unsupported for the given manifest_version', () => {
+      // Unsupported namespace in mv2.
+      let fakeSchemas = {
+        [fakeNs]: {
+          min_manifest_version: 3,
+          functions: [{ name: fakeProp }],
+        },
+      };
+
+      expect(
+        hasBrowserApi(fakeNs, fakeProp, fakeMetadata, fakeSchemas)
+      ).toBeFalsy();
+      expect(
+        hasBrowserApi(fakeNs, fakeProp, fakeMetadataV3, fakeSchemas)
+      ).toBeTruthy();
+
+      // Unsupported api method in mv2.
+      fakeSchemas = {
+        [fakeNs]: {
+          functions: [{ name: fakeProp, min_manifest_version: 3 }],
+        },
+      };
+      expect(
+        hasBrowserApi(fakeNs, fakeProp, fakeMetadata, fakeSchemas)
+      ).toBeFalsy();
+      expect(
+        hasBrowserApi(fakeNs, fakeProp, fakeMetadataV3, fakeSchemas)
+      ).toBeTruthy();
+
+      // Unsupported namespace in mv3.
+      fakeSchemas = {
+        [fakeNs]: {
+          max_manifest_version: 2,
+          functions: [{ name: fakeProp }],
+        },
+      };
+      expect(
+        hasBrowserApi(fakeNs, fakeProp, fakeMetadataV3, fakeSchemas)
+      ).toBeFalsy();
+      expect(
+        hasBrowserApi(fakeNs, fakeProp, fakeMetadata, fakeSchemas)
+      ).toBeTruthy();
+
+      // Unsupported api method in mv3.
+      fakeSchemas = {
+        [fakeNs]: {
+          functions: [{ name: fakeProp, max_manifest_version: 2 }],
+        },
+      };
+      expect(
+        hasBrowserApi(fakeNs, fakeProp, fakeMetadataV3, fakeSchemas)
+      ).toBeFalsy();
```

Comment:
nit: I'd rather use `.toEqual(false)` or `.toEqual(true)` but I don't mind (`toEqual()` only allows the value that is passed to it)

Reply:
that's a good point, I agree that a stronger assertion would be better (at least in most cases, at least the ones when we care the value to be really a boolean). I'll just change the one I added or changed directly for now (but I'm pretty sure there are many more that could also be matching this logic, we can take care of those separately).